tests/org.eclipse.passage.lic.net.tests/src/org/eclipse/passage/lic/net/tests/io/TestKeyKeeperFactory.java

```diff
@@ -0,0 +1,32 @@
+/*******************************************************************************
+ * Copyright (c) 2022 IILS mbH
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * https://www.eclipse.org/legal/epl-2.0/.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     IILS mbH (Hannes Wellmann) - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.passage.lic.net.tests.io;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+import org.eclipse.passage.lic.api.LicensingException;
+import org.eclipse.passage.lic.api.io.KeyKeeper;
+import org.eclipse.passage.lic.base.io.FileKeyKeeper;
+
+abstract class TestKeyKeeperFactory {
```

Comment:
just `TestKeyKeeper`, `Factory` is obvious in the context, which is extremely narrow here. 

Either abstract or final, means exactly what it reads: anything is either `abstract` or `final`. 
Any piece of actual implementation code defined in `acstract` class must be `final`.  

Implementation extension/overriding is prohibited in Passage core.

Redesign further.

Reply:
Done. It should fit now.